File: app/src/main/java/com/example/unsplash/model/datasource/CollectionPhotosDataSourceFactory.java
package com.example.unsplash.model.datasource;

import android.arch.lifecycle.MutableLiveData;
import android.arch.paging.DataSource;
import android.arch.paging.PageKeyedDataSource;

import com.example.unsplash.model.models.Photo;

public class CollectionPhotosDataSourceFactory extends DataSource.Factory {

    private MutableLiveData<PageKeyedDataSource<Integer, Photo>> collectionPhotoLiveDataSource = new MutableLiveData<>();
    private String id;

    public CollectionPhotosDataSourceFactory(String id) {
        this.id = id;
    }

    @Override
    public DataSource create() {
        CollectionPhotosDataSource collectionPhotosDataSource = new CollectionPhotosDataSource(id);
        collectionPhotoLiveDataSource.postValue(collectionPhotosDataSource);
        return collectionPhotosDataSource;
    }
}

File: app/src/main/java/com/example/unsplash/model/datasource/CollectionPhotosDataSourceFactory.kt
package com.example.unsplash.model.datasource

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import androidx.paging.PageKeyedDataSource
import com.example.unsplash.model.models.Photo

class CollectionPhotosDataSourceFactory(private val id: String) : DataSource.Factory<Int, Photo>() {

    private val collectionPhotoLiveDataSource = MutableLiveData<PageKeyedDataSource<Int, Photo>>()

    override fun create(): DataSource<Int, Photo> {
        val collectionPhotosDataSource = CollectionPhotosDataSource(id)
        collectionPhotoLiveDataSource.postValue(collectionPhotosDataSource)
        return collectionPhotosDataSource
    }
}
